tests/queries/tests.py

```diff
@@ -2064,6 +2064,27 @@ def test_sliced_delete(self):
         DumbCategory.objects.filter(id__in=DumbCategory.objects.order_by('-id')[1:]).delete()
         self.assertEqual(set(DumbCategory.objects.values_list('id', flat=True)), {3})
 
+    def test_distinct_ordered_sliced_subquery(self):
+        self.assertSequenceEqual(
+            DumbCategory.objects.filter(
+                id__in=DumbCategory.objects.distinct().order_by('id')[0:2],
+            ).order_by('id').values_list('id', flat=True), [1, 2]
+        )
+        # Explicit value.
+        self.assertSequenceEqual(
+            DumbCategory.objects.filter(
+                id__in=DumbCategory.objects.distinct().order_by('-id').values('id')[0:2],
+            ).order_by('id').values_list('id', flat=True), [3, 4]
+        )
+        # Annotated value.
```

Comment:
I only see the last assertion failing with the fix removed. Should the other assertions be added in a separate commit?